paper_sections/03-hierarchical_gams.Rmd

```diff
@@ -356,5 +357,40 @@ bird_mod5 <- gam(count ~ te(week, latitude, by=species, bs= c("cc", "ts"),
+
+
+```{r AIC_table, echo=F,  fig.width=4, fig.height=6,message=F, warning=F, cache=T}
+get_rsqr = function(model_list){
+  rsqr = rep(0, times= length(model_list))
+  for(i in 1:length(model_list)){
+    rsqr[i] = summary(model_list[[i]])$r.sq
+  }
+  rsqr
+}
+
+
+AIC_table = AIC(CO2_mod1,CO2_mod2, CO2_mod3, CO2_mod4, CO2_mod5,
+             bird_mod1, bird_mod2, bird_mod3, bird_mod4, bird_mod5)%>%
+  rownames_to_column(var= "Model")%>%
+  mutate(`Adjusted R squared` = get_rsqr(list(CO2_mod1,CO2_mod2, CO2_mod3, CO2_mod4,
+                                     CO2_mod5,bird_mod1, bird_mod2, bird_mod3,
+                                     bird_mod4, bird_mod5)),
+         `Adjusted R squared` = round(`R squared`, digits = 2))%>%
+  mutate_at(.vars = vars(df,AIC), .funs = funs(round,.args = list(digits=0)))
+
+
+kable(AIC_table,format ="latex", caption="AIC table comparing CO2 and bird_move model fits.")%>% #NOTE: change format to "latex" when compiling to pdf, "html" when compiling html
+  kable_styling(full_width = F)%>%
+  group_rows("A. CO2 models", 1,5)%>%
+  group_rows("B. bird_move models", 6,10)
+
+```
+
+It is important to recognize that AIC, like any function of the data, is a random variable and should be expected to have some sampling error. In cases when the goal is to select the model that has the best predictive ability, we recommend holding some fraction of the data out prior to the analysis and comparing how well different models fit that data or using k-fold cross validation as a more accurate guide to how well a given model may predict out of sample. We also strongly recommend that models are not selected based purely on AIC; instead model selection should be based on the goals of the study, computational time, and what outputs the user wants. For instance, while model 3 may fit a given dataset better than model 2, model 2 can be used to simulate functional variation for unobserved group levels, whereas this is not possible within the framework of model 3. The next section discusses these and other model fitting issues in depth. 
```

Comment:
this is nice! Maybe a wee bit more stress on the idea that we might use biology/ecology to say what's reasonable too?

Do I remember reading something about AIC random effects/smoothing parameter uncertainty?

Reply:
ah so, I think what I was thinking of was section 5 of Simon's 2016 JASA paper (Säfken + Pya paper). I'll take a look at these too though!

Reply:
Ah! That makes sense. The second paper I linked there is interesting, in that they show that conditional AIC is asymptotically equivalent to leave-one-observation-out cross-validation, and marginal AIC is asymptotically equiv. to leave-one-group-out cross-validation. Of course, Wood Pya and Säfken show that neither measure works that well...

I was trying to make a slightly different point in the text from Simon's paper, though; that, even after accounting for variability in smoothing uncertainty, any observed ΔAIC is still going to be a random variable dependent on the data; you expect to get a different ΔAIC with every new set of data (which is one of the points in Forester and Sober; I think Andrew Gelman's made the same point as well, but I can't find the reference).